pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1113,11 +1113,12 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                                     .subscriptionProperties(subscriptionProperties)
                                     .consumerEpoch(consumerEpoch)
                                     .build();
-                            if (schema != null) {
-                                return topic.addSchemaIfIdleOrCheckCompatible(schema)
-                                        .thenCompose(v -> topic.subscribe(option));
+                            if (schema != null && schema.getType() != SchemaType.AUTO_PUBLISH
+                                    && schema.getType() != SchemaType.AUTO_CONSUME) {
```

Comment:
if the client consumer sub with Schema.Type AUTO_CONSUME, the sub command arrived at the broker, the schema is null. we don't need to check this, because schema.getType() != SchemaType.AUTO_CONSUME always true

Reply:
Already fixed.